internal/storage/bucket_handle_test.go

```diff
@@ -138,3 +139,26 @@ func (t *BucketHandleTest) TestNewReaderMethodWithInvalidGeneration() {
 	AssertNe(nil, err)
 	AssertEq(nil, rc)
 }
+
+func (t *BucketHandleTest) TestDeleteObjectMethodWithValidObject() {
```

Comment:
can you testcases which tests the generation and metageneration conditions also.

Reply:
Added